File: sdk/rust/src/modules/escrow.rs
use tonic::transport::Channel;

use crate::proto::virtengine::escrow::v1::{
    query_client::QueryClient, QueryAccountRequest, QueryBalancesRequest, QueryParamsRequest,
    QueryAccountResponse, QueryBalancesResponse, QueryParamsResponse,
};
use crate::{Error, Result};

pub struct EscrowModule {
    client: QueryClient<Channel>,
}

impl EscrowModule {
    pub fn new(channel: Channel) -> Self {
        Self {
            client: QueryClient::new(channel),
        }
    }

    pub async fn balances(&mut self, owner: &str) -> Result<QueryBalancesResponse> {
        let request = QueryBalancesRequest {
            owner: owner.to_string(),
        };
        self.client
            .balances(request)
            .await
            .map(|res| res.into_inner())
            .map_err(|e| Error::Query(e.to_string()))
    }

    pub async fn account(&mut self, address: &str) -> Result<QueryAccountResponse> {
        let request = QueryAccountRequest {
            address: address.to_string(),
        };
        self.client
            .account(request)
            .await
            .map(|res| res.into_inner())
            .map_err(|e| Error::Query(e.to_string()))
    }

    pub async fn params(&mut self) -> Result<QueryParamsResponse> {
        let request = QueryParamsRequest {};
        self.client
            .params(request)
            .await
            .map(|res| res.into_inner())
            .map_err(|e| Error::Query(e.to_string()))
    }
}

File: sdk/python/virtengine/modules/escrow.py
"""Escrow module helpers."""

from __future__ import annotations

import grpc

from virtengine.escrow.v1 import query_pb2, query_pb2_grpc


class EscrowModule:
    """Escrow module queries."""

    def __init__(self, channel: grpc.aio.Channel):
        self._query = query_pb2_grpc.QueryStub(channel)

    async def balances(self, owner: str = "") -> query_pb2.QueryBalancesResponse:
        request = query_pb2.QueryBalancesRequest(owner=owner)
        return await self._query.Balances(request)

    async def account(self, address: str) -> query_pb2.QueryAccountResponse:
        request = query_pb2.QueryAccountRequest(address=address)
        return await self._query.Account(request)

    async def params(self) -> query_pb2.QueryParamsResponse:
        request = query_pb2.QueryParamsRequest()
        return await self._query.Params(request)
